### PR TITLE
Bindings and Pytests for MCMC functions

### LINKED ISSUE #376: Bindings for MCMC ops
Add Python bindings for the functions that run the MCMC relocation and covariance kernels.

### PR DESCRIPTION
**Pull request by harrism**
Closes #376 

This PR adds the MCMC relocation and add noise methods to the `GaussianSplat3D` class and adds Python bindings for them.

Additionally, the original relocation kernel was operating on plain opacities and scales, which means calling code would have to convert stored logit opacities / log scales before and after calling the function. In this PR we move the conversion (sigmoid/logit and exp/log) to and from inside the kernel to save extra torch tensor operations (separate kernels). This makes GaussianRelocation more consistent with other functions in `GaussianSplat3D` which expect logitOpacities and logScales. We also move clamping of the opacity to the range [minOpacity, 1 - FLT_EPS] inside the kernel. Also reorders the scales and opacities parameters to match other functions (scales first).

### CODE REVIEW

File: src/fvdb/GaussianSplat3d.h [hunk 1]
```diff
@@ -1189,6 +1189,24 @@ class GaussianSplat3d {
...
+    /// @brief Relocate Gaussians by adjusting opacity and scale based on replication ratio.
+    /// @param opacities logitOpacities of the Gaussians to relocate [N].
+    /// @param scales logScales of the Gaussians [N, 3] to relocate.
+    /// @param ratios Replication ratios per Gaussian [N].
+    /// @param binomialCoeffs Binomial coefficients table [nMax, nMax].
+    /// @param nMax Maximum replication ratio (size of binomial table).
+    /// @return Tuple of (logitOpacitiesNew [N], logScalesNew [N, 3])
+    std::tuple<torch::Tensor, torch::Tensor>
+    relocateGaussians(const torch::Tensor &logitOpacities, // [N]
+                      const torch::Tensor &logScales,      // [N, 3]
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/fvdb/GaussianSplat3d.h]: Fixed

File: src/fvdb/GaussianSplat3d.h [hunk 2]
```diff
@@ -1189,6 +1189,24 @@ class GaussianSplat3d {
         const bool antialias                = false,
         const int topKContributors          = 0);
 
+    /// @brief Relocate Gaussians by adjusting opacity and scale based on replication ratio.
+    /// @param opacities logitOpacities of the Gaussians to relocate [N].
+    /// @param scales logScales of the Gaussians [N, 3] to relocate.
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/fvdb/GaussianSplat3d.h]: Fixed.

File: tests/unit/test_gaussian_splat_3d.py [hunk 3]
```diff
@@ -2950,5 +2950,153 @@ def test_jagged_render_with_backgrounds(self):
...
+        return 1.0 / (1.0 + torch.exp(-100.0 * (x - 0.995)))
+
+    def test_relocate_gaussians(self):
+        if not torch.cuda.is_available():
+            self.skipTest("CUDA required for relocate_gaussians")
+
+        device = torch.device(self.device)
+        thr = 0.1
+        # Match runtime behavior in gsplat: convert logits/log-scales to probability/linear space
+        # before calling relocate_gaussians.
```
[the remark this answers is not in the document]
Comment (harrism) [reply tests/unit/test_gaussian_splat_3d.py]: Fixed.

File: src/fvdb/GaussianSplat3d.cpp [hunk 4]
```diff
@@ -1050,6 +1052,27 @@ GaussianSplat3d::renderImagesAndDepths(const torch::Tensor &worldToCameraMatrice
...
 }
 
+std::tuple<torch::Tensor, torch::Tensor>
+GaussianSplat3d::relocateGaussians(const torch::Tensor &logScales,
+                                   const torch::Tensor &logitOpacities,
+                                   const torch::Tensor &ratios,
+                                   const torch::Tensor &binomialCoeffs,
+                                   const int nMax,
+                                   const float minOpacity) {
+    return FVDB_DISPATCH_KERNEL_DEVICE(logScales.device(), [&]() {
```
Comment (matthewdcong) [inline src/fvdb/GaussianSplat3d.cpp]: This can be `FVDB_DISPATCH_KERNEL` because the `kPrivateUse1` specialization in populated.

File: src/fvdb/GaussianSplat3d.cpp [hunk 5]
```diff
@@ -1050,6 +1052,27 @@ GaussianSplat3d::renderImagesAndDepths(const torch::Tensor &worldToCameraMatrice
...
+        return detail::ops::dispatchGaussianRelocation<DeviceTag>(
+            logScales, logitOpacities, ratios, binomialCoeffs, nMax, minOpacity);
+    });
+}
+
+void
+GaussianSplat3d::addNoiseToMeans(const float noiseScale) {
+    FVDB_DISPATCH_KERNEL_DEVICE(mMeans.device(), [&]() {
```
Comment (matthewdcong) [inline src/fvdb/GaussianSplat3d.cpp]: same here

File: src/fvdb/GaussianSplat3d.cpp (hunk 4 above)
Comment (harrism) [reply src/fvdb/GaussianSplat3d.cpp]: Done.

File: src/fvdb/GaussianSplat3d.cpp (hunk 5 above)
Comment (harrism) [reply src/fvdb/GaussianSplat3d.cpp]: Done, thanks!
